assets/common/js/models/GameController.js

```diff
@@ -68,10 +76,17 @@ GameController.prototype.getPlayerWPM = function() {
 
 GameController.prototype.executeFrameActions = function() {
     // Move all the words down
-    for(word of this.words) {
+    for(let i = 0; i < this.words.length; ++i) {
+        let word = this.words[i];
         // Bonus word moves down quicker than normal words
         if(word.isBonus) word.y += this.speed + 1;
         else word.y += this.speed;
+        // Remove the word from this.words array if it reaches the bottom
+        if (word.y > this.canvas.getHeight()) { 
+            this.words.splice(i,1);
```

Comment:
Removing an element from an array in a loop requires caution. Please see https://stackoverflow.com/questions/9882284/looping-through-array-and-removing-items-without-breaking-for-loop.

Basically, when an array is spliced in a way that an element is removed, the length (this.words.length) value is changed, which will cause the loop condition (i < this.words.length) to break. This issue won't appear if there is only 1 element to be removed in the loop. However, under certain conditions, this issue can be more apparent especially if there happen to be more than 1 element to be removed in the loop. I would recommend looping in reverse to fix this issue.

Reply:
Wow yes, I didn't notice that. Big mistake of mine! :sweat_smile:. I will fix it :smile:. Thanks for noticing this. I appreciate it. :heart:.